lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -31,7 +35,17 @@ export class BrowserStorage implements IWindowStorage<string> {
     }
 
     setItem(key: string, value: string): void {
-        this.windowStorage.setItem(key, value);
+        try{
+            this.windowStorage.setItem(key, value);
+        }
+        catch(e){
+            if(e.code === BrowserAuthErrorMessage.chromiumStorageException.code || e.code === BrowserAuthErrorMessage.firefoxStorageException.code)
+            {
+                this.logger.error("Could not access browser storage. This may be caused by exceeding the quota.");
```

Comment:
Should we attempt to clean up some of our own entries here and retry?

Reply:
In fact I kind of think we should. Not doing so may result in other unintended errors or behaviors, for example temporary cache entries being missing at the end of the flow when they're needed or `getAllAccounts` returning an empty array after a  successful login. I also wonder if there might be a way we can signal to the caller that storage was full so they can do further cleanup on their own, perhaps by using the event API?

Reply:
Have modified to include event handler to emit event which can signal the user to do clean on their own, as for the first part, I am now calling clear() to clear all the msal items in storage, do you have any suggestions on how to retry after clearing the storage?

Reply:
Can we prompt the user to use a custom storage option in this scenario?

Reply:
I think it would be great if we could maintain cache eviction policy for our own keys. Cleaning up all of them at once in the middle of the flow may produce unexpected results. Please correct me if I am wrong.

Reply:
@konstantin-msft I agree, based on discussion with @tnorling yesterday, it would be better to have a design document to discuss the eviction policies and implications on all flows.Thanks guys for feedback!